tests/autobahn/test_autobahn.py

```diff
@@ -0,0 +1,163 @@
+import errno
+import json
+import shutil
+import socket
+import subprocess
+import sys
+import time
+from pathlib import Path
+from typing import Any, Dict, List
+
+import pytest
+
+HOST = "localhost"
+PORT = 9001
+
+
+@pytest.fixture(scope="module", autouse=True)
+def create_report_directory(request: Any) -> None:
+    path = Path(f"{request.fspath.dirname}/reports")
+    if path.is_dir():
+        shutil.rmtree(path)
+    path.mkdir()
+
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_venv() -> None:
+    if not Path("tests/autobahn/autobahntestsuite-env").exists():
+        print("Creating Python 2.7 environment and installing autobahntestsuite")
+        subprocess.check_call(
+            ["virtualenv", "-p", "python2.7", "tests/autobahn/autobahntestsuite-env"]
+        )
+        subprocess.check_call(
+            [
+                "tests/autobahn/autobahntestsuite-env/bin/pip",
+                "install",
+                "autobahntestsuite>=0.8.0",
+            ]
+        )
+
+
+def wait_for_server(host: str, port: int) -> None:
+    while True:
+        sock = socket.socket()
+        try:
+            sock.connect((HOST, PORT))
+        except OSError as sock_err:
+            if sock_err.errno == errno.ECONNREFUSED:
+                time.sleep(0.01)
+            else:
+                raise
+        else:
+            return
+        finally:
+            sock.close()
+
+
+def get_failed_tests(report_path: str, name: str) -> List[Dict[str, Any]]:
+    with open(Path(f"{report_path}/index.json")) as f:
+        result_summary = json.load(f)[name]
+    failed_messages = []
+    PASS = {"OK", "INFORMATIONAL"}
+    for results in result_summary.values():
+        if results["behavior"] not in PASS or results["behaviorClose"] not in PASS:
+            with open(Path(f"{report_path}/{results['reportfile']}")) as f:
```

Comment:
When working with pathlib, don't concatenate strings with an os-specific separator (it's different under windows, for example.

Reply:
Thank you for the tip 🙏.

Reply:
The new code is definitely better than using an f-string, but I believe it's actually fine using `/` in strings for pathlib (it was a problem back in the `os.path` days). The docs do it in many examples, and it is automatically converted to the system separator:
```
>>> str(pathlib.PureWindowsPath("a/b"))
'a\\b'
>>> str(pathlib.PurePosixPath("a/b"))
'a/b'
```

Even when constructing it programmatically, it uses `/` in the `repr` version as a universal representation:
```
>>> p = pathlib.PureWindowsPath("a") / "b"
>>> p
PureWindowsPath('a/b')
>>> str(p)
'a\\b'
```